From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Akash Yadav <itsaky01@gmail.com>
Date: Sat, 17 May 2025 20:47:20 +0000
Subject: [PATCH] Android Components - Allow overriding the build date

Adds support for overriding the build date via the `IRONFOX_BUILD_DATE_OVERRIDE` environment variable.

Signed-off-by: Akash Yadav <itsaky01@gmail.com>
---
diff --git a/mobile/android/android-components/plugins/config/src/main/java/ConfigPlugin.kt b/mobile/android/android-components/plugins/config/src/main/java/ConfigPlugin.kt
index f82613fdd6..cf55ee61b5 100644
--- a/mobile/android/android-components/plugins/config/src/main/java/ConfigPlugin.kt
+++ b/mobile/android/android-components/plugins/config/src/main/java/ConfigPlugin.kt
@@ -60,7 +60,13 @@ object Config {
     private val fennecBaseVersionCode by lazy {
         val format = SimpleDateFormat("yyyyMMddHHmmss", Locale.US)
         val cutoff = format.parse("20141228000000")
-        val build = Date()
+        var build = Date()
+
+        val buildDate = System.getenv("IRONFOX_BUILD_DATE_OVERRIDE")
+        if (!buildDate.isNullOrBlank() && buildDate != "null") {
+            println("[IronFox] Overriding build date to: '$buildDate'")
+            build = Date.from(java.time.Instant.parse(buildDate))
+        }
 
         Math.floor((build.time - cutoff.time) / (1000.0 * 60.0 * 60.0)).toInt()
     }
-- 
